### PR TITLE
ORC-715: Add MapReduce test case

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

This pr aims to recover a test coverage which is removed at
https://github.com/apache/orc/commit/db62ea5730797b046a3b7d5d1a3b026b08921209

Refer Hadoop example and previous mapreduce tests.
https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestMapReduce.java
https://github.com/apache/orc/pull/612/files#diff-15999a33f815f28f3251acabae2a16dd2f1114d8c73e6fab26c207c08884ce68

BTW This pr bypasses the bug in older versions of hadoop-common by configuring `mapreduce.job.output.key.comparator.class`.  Fix [ORC-964 ](https://issues.apache.org/jira/projects/ORC/issues/ORC-964?filter=allopenissues) issue.

### Why are the changes needed?

Recover a test coverage

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for making this, @guiyanakuang .
Mostly this PR looks good. The only difference I noticed is that this is testing vectorized code path instead.  Is this change intentional?

File: java/mapreduce/src/test/org/apache/orc/mapreduce/TestMrUnit.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+      .fromString("struct<one:struct<x:int,y:int>,two:struct<z:string>>");
+
+  private static final TypeDescription OUT_SCHEMA = TypeDescription
+      .fromString("struct<first:struct<x:int,y:int>,second:struct<z:string>>");
+
+  static {
+    OrcConf.MAPRED_SHUFFLE_KEY_SCHEMA.setString(CONF, "struct<x:int,y:int>");
+    OrcConf.MAPRED_SHUFFLE_VALUE_SCHEMA.setString(CONF, "struct<z:string>");
+    OrcConf.MAPRED_OUTPUT_SCHEMA.setString(CONF, OUT_SCHEMA.toString());
+    CONF.set("mapreduce.job.output.key.comparator.class", OrcKeyComparator.class.getName());
```
Comment (dongjoon-hyun) [inline java/mapreduce/src/test/org/apache/orc/mapreduce/TestMrUnit.java:74]: Thank you for adding this. Could you add a comment referring ORC-964?

**Comment by guiyanakuang** [comment]
> Thank you for making this, @guiyanakuang .
> Mostly this PR looks good. The only difference I noticed is that this is testing vectorized code path instead. Is this change intentional?

This is not intentional. I'll change pr later to use OrcMapreduceRecordWriter and OrcMapreduceRecordReader instead.

File: java/mapreduce/src/test/org/apache/orc/mapreduce/TestMrUnit.java (hunk 1 above)
Comment (guiyanakuang) [reply java/mapreduce/src/test/org/apache/orc/mapreduce/TestMrUnit.java:74]: OK 👌

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @guiyanakuang .

cc @williamhyun because this will be a part of ORC-716 .

**Comment by dongjoon-hyun** [comment]
Merged to main/1.7.

**Pull request merged**